### ISSUE TITLE
pushing the resources to the cloud: Cannot read property 'join' of null

### ISSUE THREAD

**Issue by Tom-RulesCube** [aws-amplify/amplify-cli#4688]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**

After an attempt at merging new work from a feature branch and separate amplify env ...
back to the the dev branch with it's own amplify env:

```
amplify push
...
✖ An error occurred when pushing the resources to the cloud

Cannot read property 'join' of null
An error occurred during the push operation: Cannot read property 'join' of null
```

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system

4.22.0

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable

```
# from feature branch/env
amplify env checkout devenv
amplify push
git add .
git commit ...
# other dev working on branch
amplify pull 
# resolve some regressions b/c amplify and git ... 
amplify status
# looks good ... changes to push
amplify push
? Are you sure you want to continue? Yes
✖ An error occurred when pushing the resources to the cloud

Cannot read property 'join' of null
An error occurred during the push operation: Cannot read property 'join' of null
```

**Expected behavior**

Expected to deploy the dev environment

**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. Mac/Windows/Ubuntu]
 - Node Version. You can use `node -v` to check the node version on your system

Mac OS Catalina 10.15.5

**Additional context**

The development branch had changes to the schema (new models) and changes to llambda functions

The feature branch also had additions to the schema (new models) and changes to lambda functions

Both environments were working and tested. We wanted to merge the feature back into the development branch. Tbh we're struggling with this part of Amplify and this is the 3rd or 4th attempt we've made at a clean merge process.

To merge we did an `amplify env checkout devenv` then an `amplify push` as I understand is the process(?)

We stashed the code changes, then switched to the dev branch / env and did `amplify pull` 

Used the stashed changes to re-apply code changes. And did a manual merge on the graphql.schema

Then this `amplify push` ... which failed.

**Comment by akshbhu** [comment]
Hi @Tom-RulesCube 

There is pretty neat documentation for working with multienv. It seems to cover the use case you are describing.Take a look at here: https://docs.amplify.aws/cli/teams/overview

Let me know if it works for you?

**Comment by Tom-RulesCube** [comment]
I will take a look at that again, we did start out with that, but a number of times ended up with a zombie environment that could not be rescued. I realize the tools have been updated continually. But it would be great to find out what was bothering push ... we're not really doing anything controversial, from the documentation point of view.

Is there any way to get it give more explicit diagnostics? I'm happy to go digging in code if I can find out what it's choking on.

**Comment by akshbhu** [comment]
Hi @Tom-RulesCube 

May I know what categories you added so that I can reproduce this and if its possible for you send the amplify folder to amplify-cli@amazon.com with the issue no. marked as header in the mail.

**Comment by Tom-RulesCube** [comment]
We've moved on -- basically reverted and are doing a careful merge ... I will check out that instance of our amplify folder and send it.

**Comment by gildniy** [comment]
Got the same issue and had to pull the app afresh as @Tom-RulesCube suggested.

**Comment by bmurillop** [comment]
I recently got it. As suggested before I started with the docs to work with multiple envs, a different points got zombie envs that I had to ditch and start over again. In this try I created an angular app, initialized the amplify repo, added auth and tested then tried and it worked, then added API to the backend and building it throws this error ... I havent even modified the sample angular app.

**Comment by dodtsair** [comment]
I have emailed my amplify folder.  

I am not sure how I produced the issues.  I can say lots of things went wrong

1) When adding and removing lambda functions and models the Cloudformation stack corrupted.  I didn't bother trying to recover it.  Instead (probably not a good idea) I deleted the parent stack
2) I used amplify push to create it again, but that failed
3) Checked in the code and used amplify delete
4) Checked out the code ran amplify push
5) That failed so I ran amplify init
6) Then tried amplify push again
5) Got some s3 missing error, and promise rejection unhandled error 
6) I moved team-provider-info.json to team-provider-info.json.old
7) Then ran amplify push again

Got error:
```
gold:react-amplified michael.power$ aws s3 ls
2020-09-18 16:30:42 amplify-reactamplified-dev-163031-deployment
gold:react-amplified michael.power$ amplify push
✔ Successfully pulled backend environment dev from the cloud.

Current Environment: dev

| Category | Resource name          | Operation | Provider plugin   |
| -------- | ---------------------- | --------- | ----------------- |
| Api      | myapi                  | Create    | awscloudformation |
| Auth     | reactamplified6329fb03 | Create    | awscloudformation |
| Hosting  | amplifyhosting         | Create    | awscloudformation |
| Function | addingfunction         | Create    | awscloudformation |
| Function | createwatch            | Create    | awscloudformation |
| Function | getwatchlist           | Create    | awscloudformation |
? Are you sure you want to continue? Yes
✖ An error occurred when pushing the resources to the cloud

Cannot read property 'join' of null
An error occurred during the push operation: Cannot read property 'join' of null
```

**Comment by idanlo** [comment]
anyone know why this is happening? I am getting this error on my environment where I have only modified code of a couple of functions. This is an environment with many resources so I would like to get a solution that is not "delete and recreate the environment" please

**Comment by kaustavghosh06** [comment]
@idanlo Do you have functions configured in your project? You can run `amplify function build` to get a better stack trace of what the error is. Could you please run that and let us know the stack-trace?

**Comment by idanlo** [comment]
@kaustavghosh06 after deleting the amplify folder locally and doing init again it worked. Weird

**Comment by yuth** [comment]
https://github.com/aws-amplify/amplify-cli/pull/6126 has the fix. This has been released in Amplify CLI v4.40.1. Feel free to comment here if the issue still persists
